### DISCUSSION TITLE
How to host a static site at Codeberg Pages?

### DISCUSSION THREAD

**Discussion by nevillepark** [datenstrom/community#940] [Ask a question] [answered]
Hi, I'm experiencing something odd when generating a static version of a Yellow site with the Generate extension. 

When there is only one page in a folder, an `index.html` file is generated, and this works fine. However, any other pages in the folder are not generated with an `.html` on the end. As a result, the browser tries to download the HTML file instead of displaying it. (This happens in Firefox, GNOME Web, and lynx.) Manually adding `.html` to the filename fixes the problem.

For example, the files that are generated by default with the Blog extension are:

```
blog/
├ 2020-04-07-blog-example-page.md
├ 2020-12-06-made-for-people.md
└ page.md
```

When I run `php yellow.php generate`, what is produced is

```
blog/
├ author:datenstrom/
│ └ index.html 
├ tag:example/
│ └ index.html 
├ blog-example-page
├ index.html
└ made-for-people
```

Here's [a freshly set-up example site](https://nev.codeberg.page/yellow-test). You can see the Yellow install directory on [the "main" branch](https://codeberg.org/nev/yellow-test/src/branch/main) and the generated static site files on [the "pages" branch](https://codeberg.org/nev/yellow-test/src/branch/pages).

If you go to [the blog index page](https://nev.codeberg.page/yellow-test/blog/) and try to click on [the first blog post](https://nev.codeberg.page/yellow-test/blog/made-for-people), the browser will try to download the HTML file. 

However, renaming the file to have an `.html` extension fixes it. See for example the `/about/` directory, where I changed `more-about-stuff` to `more-about-stuff.html`; [the "More About Page" link on the main About page](https://nev.codeberg.page/yellow-test/about/) should properly load [the page it points to](https://nev.codeberg.page/yellow-test/about/more-about-stuff.html).

At first I thought it was a problem with the customized theme files I was using, or some setting I had changed, but it happens even with the default Stockholm theme and a fresh, updated install. No errors show up in [the website log](https://codeberg.org/nev/yellow-test/src/branch/main/system/extensions/yellow-website.log) or when running the generate command in debug mode.

- Yellow version: 0.8.23, Core 0.8.125
- PHP version: 8.1.2-1ubuntu2.14
- Browsers I've tried this in: Firefox 123.0, GNOME Web 45.2, lynx 2.9.0dev.10
- OS: elementary OS 7.1, based on Ubuntu 22.04.3 LTS

Can anyone else replicate this? What other information would help diagnose the problem?

**Comment by schulle4u** [comment, marked as answer]
I think this is a similar problem to the one discussed in the context of datenstrom/community#367. Codberg pages apparently doesn't recognize the correct mimetype of sub-pages, thus sending them as `application/octed-stream` in this case. I suggest to check their pages documentation and find out if you can modify the default mimetype for specific folders (e.g. everything under `/content` should be treated as text/html).

**Reply by nevillepark** [reply]
Thanks so much, this pointed me in the right direction—there's some related discussions in the Codeberg Pages server issue tracker. I'll see if there's a way to configure it properly.

**Reply by schulle4u** [reply]
You're welcome. Let us know if you find a solution. 🙂
